### PR TITLE
feat(demo): supabase demo

### PR DESCRIPTION
**Pull request by sogunshola**
---
name: 'Pull request'
about: A new pull request
---

### Checklist

- [X] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [X] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [X] I have made corresponding changes to the documentation
- [X] I have updated Typescript definitions when needed

### Issue Description

Related issue: #`FILL_THIS_OUT`

### CODE REVIEW

File: demos/demo-supabase-auth/src/auth/MoralisAuthAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        authData.domain = data.domain;
+        authData.expirationTime = data.expirationTime;
+        authData.notBefore = data.notBefore;
+        authData.resources = data.resources;
+        authData.statement = data.statement;
+        authData.uri = data.uri;
+        return;
+      }
+
+      // @ts-ignore (see note at top of file)
```
Comment (b4rtaz) [inline demos/demo-supabase-auth/src/auth/MoralisAuthAdapter.ts:27]: I don't see any note at top.

**Review by b4rtaz — CHANGES_REQUESTED** [review]

File: demos/demo-supabase-auth/src/auth/MoralisAuthAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        authData.expirationTime = data.expirationTime;
+        authData.notBefore = data.notBefore;
+        authData.resources = data.resources;
+        authData.statement = data.statement;
+        authData.uri = data.uri;
+        return;
+      }
+
+      // @ts-ignore
+      throw new Parse.Error(Parse.Error.OBJECT_NOT_FOUND, 'Moralis auth failed, invalid data');
```
Comment (ErnoW) [inline demos/demo-supabase-auth/src/auth/MoralisAuthAdapter.ts:28]: There is no Parse is this project.

Is this auth-adapter used in this demo, as this file was specifically needed in the parse-server demo.

File: demos/demo-supabase-auth/src/auth/authService.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  return message;
+}
+
+export interface VerifyMessage {
+  network: string;
+  signature: string;
+  message: string;
+}
+
+export async function verifyMessage({ network, signature, message }: VerifyMessage) {
```
Comment (ErnoW) [inline demos/demo-supabase-auth/src/auth/authService.ts:45]: Where are we actually verifying the signature, that is not clear to me

File: demos/demo-supabase-auth/src/middlewares/errorHandler.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    const status = typeof error.details?.status === 'number' ? error.details?.status : 500;
+    let errorMessage = error.message || 'Unknown error';
+
+    if (errorResponseData) {
+      // Handle MoralisError
+      if (errorResponseData && errorResponseData?.message) {
+        errorMessage = `${errorResponseData?.name ? `${errorResponseData.name}: ` : ''}${errorResponseData.message}`;
+      } else if (errorResponseData.error) {
+        // Handle ParseError
```
Comment (ErnoW) [inline demos/demo-supabase-auth/src/middlewares/errorHandler.ts:18]: We have no ParseErrors in this project

**Review by ErnoW — CHANGES_REQUESTED** [review]

File: .changeset/slow-feet-tell.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'demo-subabase-auth': major
```
[the remark this answers is not in the document]
Comment (sogunshola) [reply .changeset/slow-feet-tell.md:2]: Yes this helps.. thanks for this

**Review by ErnoW — APPROVED** [review]

**Pull request merged**